src/looklike.ml

```diff
@@ -0,0 +1,150 @@
+(** Looklike allows you to find recurrent patterns in some lists *)
+
+
+(** Module type which allows comparing elements *)
+module type EqType = 
+sig
+  type t
+  val equal : t -> t -> bool
+  val nil : t 
+  val get_str : t -> string
+end
```

Comment:
`nil` and `get_str` seem unneeded to equip a type with a comparison. Can you explain why they are here?

Reply:
Nil is used when you have to compare a type t. It's the equivalent of neutral element for type t.
get_str is just a way to be sure this is a printable element.

Reply:
What is the meaning of a neutral for a comparison?

Reply:
This is not a signature for a ComparableType then. It is a PrintableAndComparableType, right?

Use `include` to include both signatures in a single signature.

Reply:
Neutral is maybe a wrong word... It's an equivalent of None or empty. Sometimes in the parsing process you want to avoid newline or '\n' or other stuff, depending on of your t type. Otherwise, it matches alone new lines and inserts reference(s) everywhere.

Reply:
You say that I must use something like this ?
```ocaml
module TrimString  = 
struct
  type t = string
  let equal str1 str2 = 
    (String.trim str1) = (String.trim str2)
  include PrintString
  let nil = ""
end
```